File: XPile 3-17/Programs/test_slice.py
from dataclasses import dataclass, field

def __ul_sort(arr, n): arr[:n] = sorted(arr[:n])

class __ul_Exception(Exception):
    def __init__(self, type_name, message):
        self.type_name = type_name
        self.message = message
        super().__init__(message)


def main():
    greeting = 'Hello World'
    hello = greeting[0:5]
    print(hello)
    world = greeting[6:11]
    print(world)
    nums = [10, 20, 30, 40, 50]
    middle = nums[1:4]
    print(middle[0])
    print(middle[1])
    print(middle[2])
    h = greeting[0:1]
    print(h)
    return 0


if __name__ == "__main__":
    main()

File: XPile 3-17/Programs/test_slice.js

function main() {
    let greeting = "Hello World";
    let hello = greeting.slice(0, 5);
    console.log(hello);
    let world = greeting.slice(6, 11);
    console.log(world);
    let nums = [10, 20, 30, 40, 50];
    let middle = nums.slice(1, 4);
    console.log(middle[0]);
    console.log(middle[1]);
    console.log(middle[2]);
    let h = greeting.slice(0, 1);
    console.log(h);
    return 0;
}

function __ul_sort(arr, n) { const sub = arr.slice(0, n).sort((a, b) => a < b ? -1 : a > b ? 1 : 0); for (let i = 0; i < n; i++) arr[i] = sub[i]; }

main();
